### PR TITLE
feat(apps & services):- Post autosetup

### PR DESCRIPTION
**Pull request by qxz2mqe**
## Description
Added technicalUserPermissions and clienturl in response body

## Why

to support the FE/user with additional information

## Issue

https://jira.catena-x.net/browse/CPLP-2665

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)

- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/marketplace/Offers.Library/Service/OfferSetupService.cs [hunk 1]
```diff
@@ -281,7 +281,7 @@ private async Task<(string clientId, Guid iamClientId)> CreateClient(string offe
         var clientId = await _provisioningManager.SetupClientAsync(redirectUrl, offerUrl, userRoles, enabled)
             .ConfigureAwait(false);
         var iamClient = _portalRepositories.GetInstance<IClientRepository>().CreateClient(clientId);
-        return (clientId, iamClient.Id);
+        return (clientId, iamClient.Id, offerUrl);
```
Comment (ntruchsess) [inline src/marketplace/Offers.Library/Service/OfferSetupService.cs:284]: What shall this be good for? there is no point in adding an unchanged parameter of a method to the return-values.

File: src/marketplace/Offers.Library/Service/OfferSetupService.cs [hunk 2]
```diff
@@ -113,8 +113,8 @@ public async Task<OfferAutoSetupResponseData> AutoSetupOfferAsync(OfferAutoSetup
         ClientInfoData? clientInfoData = null;
         if (offerTypeId == OfferTypeId.APP)
         {
-            var (clientId, iamClientId) = await CreateClient(data.OfferUrl, offerDetails.OfferId, true, userRolesRepository).ConfigureAwait(false);
-            clientInfoData = new ClientInfoData(clientId);
+            var (clientId, iamClientId, clientUrl) = await CreateClient(data.OfferUrl, offerDetails.OfferId, true, userRolesRepository).ConfigureAwait(false);
```
Comment (ntruchsess) [inline src/marketplace/Offers.Library/Service/OfferSetupService.cs:116]: clientUrl is the unchanged value of data.OfferUrl. There is no point in doing this assignment via a call to CreateClient.

**Review by ntruchsess — CHANGES_REQUESTED** [review]

**Comment by ntruchsess** [comment]
> Please delete options that are not relevant.
> 
>     * [ ]  I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
> 
>     * [ ]  I have performed a self-review of my own code
> 
>     * [ ]  I have successfully tested my changes locally
> 
>     * [ ]  I have added tests that prove my changes work
> 
>     * [ ]  I have checked that new and existing tests pass locally with my changes
> 
>     * [ ]  I have commented my code, particularly in hard-to-understand areas

please make sure you mark the options that you confirm your PR does comply with. To do so replace the space enclosed by the square-brackets by the character 'X'. Like so
`* [X]  I have performed a self-review of my own code`
* [X]  I have performed a self-review of my own code
github will render this as a checked checkbox.

File: src/marketplace/Offers.Library/Service/OfferSetupService.cs (hunk 1 above)
Comment (qxz2mqe) [reply src/marketplace/Offers.Library/Service/OfferSetupService.cs]: fixed

File: src/marketplace/Offers.Library/Service/OfferSetupService.cs (hunk 2 above)
Comment (qxz2mqe) [reply src/marketplace/Offers.Library/Service/OfferSetupService.cs]: fixed

**Review by ntruchsess — DISMISSED** [review]

**Review by ntruchsess — APPROVED** [review]

**Pull request merged**